mypy/checkexpr.py

```diff
@@ -4273,34 +4287,41 @@ def check_boolean_op(self, e: OpExpr, context: Context) -> Type:
         elif e.op == "or":
             left_map, right_map = self.chk.find_isinstance_check(e.left)
 
+        left_impossible = left_map is None or any(
+            isinstance(get_proper_type(v), UninhabitedType) for v in left_map.values()
```

Comment:
This expression is duplicated quite a lot. Would it make sense to move this to a helper function?